safekeeper/src/timeline.rs

```diff
@@ -191,79 +190,33 @@ impl SharedState {
         self.wal_backup_active
     }
 
-    /// start/change walsender (via callmemaybe).
-    fn callmemaybe_sub(
+    /// Activate timeline's walsender: start/change timeline information propagated into etcd for further pageserver connections.
+    fn activate_walsender(
```

Comment:
This can be removed altogether, with new model don't need anything on safekeeper side except for pushing to etcd.

Reply:
I would leave it right now in this PR, since there's some logic in safekeeper, related to stopping walsender via setting this field to `None` in the timeline (hence, it stops sending it to etcd and pageserver does not see it in the list of candidates): https://github.com/neondatabase/neon/blob/0e1bd57c533165dbe4bead8fa23baefa09c97b82/safekeeper/src/send_wal.rs#L318

I agree that etcd subscription should be removed here instead, but let's do that separately.

Reply:
https://github.com/neondatabase/neon/issues/1844